src/components/AddHotel.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { getAllHotelsAsync } from '../redux/hotel/hotel';
+
+const AddHotel = () => {
+  const [enteredName, setEnteredName] = useState('');
+  const [enteredDescription, setEnteredDescription] = useState('');
+  const [enteredImage, setEnteredImage] = useState(null);
+  const [enteredRating, setEnteredRating] = useState(0);
+  const [enteredOption, setEnteredOption] = useState('');
+
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const nameChangeHandler = (event) => {
+    setEnteredName(event.target.value);
+  };
+
+  const descriptionChangeHandler = (event) => {
+    setEnteredDescription(event.target.value);
+  };
+
+  const ratingChangeHandler = (event) => {
+    setEnteredRating(event.target.value);
+  };
+
+  const imageChangeHandler = (event) => {
+    setEnteredImage(event.target.files[0]);
+  };
+
+  const cityChangeHandler = (event) => {
+    setEnteredOption(event.target.value);
+  };
+
+  const token = useSelector((state) => state.login.token);
+  const cities = useSelector((state) => state.city.all);
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+
+    const formData = new FormData();
+    formData.append('name', enteredName);
+    formData.append('description', enteredDescription);
+    formData.append('rating', enteredRating);
+    formData.append('image', enteredImage);
+    formData.append('city_id', enteredOption);
+
+    fetch('https://hotelzilla-api.herokuapp.com/api/hotels', {
+      method: 'POST',
+      headers: {
+        Authorization: token,
+      },
+      body: formData,
+    })
+      .then(() => {
+        dispatch(getAllHotelsAsync());
+        navigate('/', { replace: true });
+      })
+      .catch((error) => console.log(error));
+  };
```

Comment:
- [ ] [Optional] I strongly suggest you move the fetch call to a redux async thunk and control the server response in order for you to return success, pending and rejected messages based on the fetch status. This will improve user experience and is going to give you the ability to handle redirection, clear the form and show server alerts based on server response, right now the user is blind to what's happening after creating a new hotel.

Reply:
Hi @williamrolando88 I tried to use async thunk with this but there is a problem with the attached image when I use async thunk, so for now, I created an API helper and used it here instead of calling it directly.